Experiences/Experiences/View Controllers/AddExperienceViewController.swift

```diff
@@ -0,0 +1,346 @@
+//
+//  AddExperienceViewController.swift
+//  Experiences
+//
+//  Created by Dillon P on 3/22/20.
+//  Copyright © 2020 Lambda iOSPT3. All rights reserved.
+//
+
+import UIKit
+import Photos
+import CoreImage
+import CoreImage.CIFilterBuiltins
+import AVFoundation
+
+class AddExperienceViewController: UIViewController {
+    
+    
+    // MARK: - Outlets
+    
+    @IBOutlet weak var titleTextField: UITextField!
+    @IBOutlet weak var imageView: UIImageView!
+    @IBOutlet weak var chooseImageButton: UIButton!
+    @IBOutlet weak var recordAudioButton: UIButton!
+    
+    
+    //MARK: - View Set-Up
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        audioPlayer = AVAudioPlayer()
+        titleTextField.delegate = self
+        
+    }
+    
+    private func updateViews() {
+        recordAudioButton.isSelected = isRecording
+    }
+    
+    // MARK: Model Controller Properties & Delegate Methods
+    
+    var experienceController: ExperienceController?
+    var delegate: MapViewReloadDelegate?
+    
+    
+    // MARK: - Image Properties
+    
+    var originalImage: UIImage? {
+        didSet {
+            imageView.image = makeImageBW(byFiltering: originalImage!)
+            imageData = imageView.image?.jpegData(compressionQuality: 0.1)
+        }
+    }
+    
+    var imageData: Data?
+    
+    
+    //MARK: - Audio Properties
+    
+    var audioPlayer: AVAudioPlayer? {
+        didSet {
+            guard let audioPlayer = audioPlayer else { return }
+            audioPlayer.delegate = self
+            updateViews()
+        }
+    }
+    
+    var audioRecorder: AVAudioRecorder?
+    var recordingURL: URL?
+    
+    var isRecording: Bool {
+        audioRecorder?.isRecording ?? false
```

Comment:
great use of a nil coalescing here. putting it on one line instead of an if else statement is a boss move.